File: Keepr/Models/VaultKeep.cs
namespace Keepr.Models
{
    public class VaultKeep
    {
        public int Id { get; set; }
        public string CreatorId { get; set; }

        public int VaultId { get; set; }

        public int KeepId { get; set; }

          public Profile Creator { get; set; }

          public Vault Vault { get; set; }
    }

    public class VaultKeepViewModel : Keep
    {
    //  public int KeepId { get; set; }

    //  public int VaultId { get; set; }

     public int VaultKeepId { get; set; }

      // public Vault Vault { get; set; }

      // public Keep Keep { get; set; }
}
}

File: Keepr.client/src/Models/VaultKeep.js
export class VaultKeep {
  constructor(data = {}){
    this.vaultKeepId = data.id;
    this.creatorId = data.creatorId;
    this.createdAt = data.createdAt;
    this.updatedAt = data.updatedAt;
    this.creator = data.creator || {};
    this.keepId = data.keepId;
    this.vaultId = data.vaultId;
    this.name = data.name;
    this.img = data.img;
  }
}